app/components/market/submit/submit-market-application.component.ts

```diff
@@ -1,208 +1,171 @@
         });
     }
-
-    public createForms() {
-        let parameterKeys = Object.keys(this.jobTemplate.parameters);
+    private createForms() {
+        let jobParameters = [];
+        let poolParameters = [];
+        if (this.jobTemplate && this.jobTemplate.parameters){
+            jobParameters = Object.keys(this.jobTemplate.parameters);
+        }
+        if (this.poolTemplate && this.poolTemplate.parameters){
+            poolParameters = Object.keys(this.poolTemplate.parameters);
+        }
         let fg = {};
-        for (let key of parameterKeys) {
+
+        for (let key of jobParameters) {
             if ("defaultValue" in this.jobTemplate.parameters[key]) {
                 const defaultValue = String(this.jobTemplate.parameters[key].defaultValue);
                 fg[key] = new FormControl(defaultValue);
             } else {
                 fg[key] = new FormControl();
             }
         }
-        this.jobFormGroup = this.formBuilder.group(fg);
-
-        this.form = this.formBuilder.group({
-            job: this.jobFormGroup,
-            pool: this.pickedPool,
-        });
-        if (this._formValue) {
-            this.form.setValue(this._formValue);
+        for (let key of poolParameters) {
+            if ("defaultValue" in this.poolTemplate.parameters[key]) {
+                const defaultValue = String(this.poolTemplate.parameters[key].defaultValue);
+                fg[key] = new FormControl(defaultValue);
+            } else {
+                fg[key] = new FormControl();
+            }
         }
-        this.form.valueChanges.debounceTime(400).distinctUntilChanged().subscribe((newFormValue) => {
-            this.router.navigate([], {
-                relativeTo: this.route,
-                queryParams: {
-                    formParams: JSON.stringify(newFormValue),
-                },
-            });
-        });
-    }
-
-    public getContainerFromFileGroup(fileGroup: string) {
-        return fileGroup && `fgrp-${fileGroup}`;
+        this.form = new FormGroup(fg);
     }
 
-    @autobind()
-    public submit() {
-        this.error = null;
-        const jobParams = this.jobFormGroup.value;
-        // RPC takes in Template JSON object and Parameter JSON object
-        const obs = this.pythonRpcService.callWithAuth("submit-ncj-job", [this._buildJobTemplate(), jobParams]);
+    private runJobWithPool(expandedPoolTemplate, jobInputs){
+        delete expandedPoolTemplate.id;
+        this.jobTemplate.job.properties.poolInfo = {
+            autoPoolSpecification: {
+            autoPoolIdPrefix: "jobname",
+            poolLifetimeOption: "job",
+            keepAlive: false,
+            pool: expandedPoolTemplate,
+            },
+        };
+        const obs = this.pythonRpcService.callWithAuth("submit-ncj-job", [this.jobTemplate, jobInputs]);
         obs.subscribe({
-            next: (data) => this._redirectToJob(data),
+            next: (data) => this._redirectToJob(),
```

Comment:
Why did you remove the 'data' object from this call? The expected design is to navigate to the job we just created. Looking at the code below, you no longer seem to have a reference to the job id and can only navigate to the jobs list UI.

Also, a second comment. We should have a type on data, eg.
```ts
next: (data: myObj) => this._redirectToJob()
```